pkg/k8s/watchers/cilium_node.go

```diff
@@ -69,7 +69,8 @@ func (k *K8sWatcher) ciliumNodeInit(ciliumNPClient client.Clientset, asyncContro
 							valid = true
 							isLocal := k8s.IsLocalCiliumNode(ciliumNode)
 							if oldCN.DeepEqual(ciliumNode) &&
-								comparator.MapStringEquals(oldCN.ObjectMeta.Labels, ciliumNode.ObjectMeta.Labels) {
+								comparator.MapStringEquals(oldCN.ObjectMeta.Labels, ciliumNode.ObjectMeta.Labels) &&
+								comparator.MapStringEquals(oldCN.ObjectMeta.Annotations, ciliumNode.ObjectMeta.Annotations) {
```

Comment:
It might be unclear for someone reading this why we need to compare annotations - could you add a comment explaining the context?

Reply:
Updated the commit with comment.